src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+const getUser = (userId: number): User | null => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+export const visibleTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [textTitle, addTitle] = useState('');
+  const [userId, changeUser] = useState(0);
+  const [titleError, showTitleError] = useState(false);
+  const [userError, showUserError] = useState(false);
+  const maxId = Math.max(...visibleTodos.map(todo => todo.id));
+
+  const titleHendler = (e:React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Please look at this article
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364